src/sync/db.rs

```diff
@@ -139,3 +136,68 @@ pub fn fmt_tool_names<F: FnMut(&String) -> String>(fmt_tool: F) -> String {
         .collect::<Vec<String>>()
         .join("\n")
 }
+
+struct ToolBTreeMap {
+    _inner_map: BTreeMap<String, ToolInfo>,
+}
+
+impl ToolBTreeMap {
+    fn new() -> Self {
+        Self {
+            _inner_map: BTreeMap::new(),
+        }
+    }
+
+    fn insert(&mut self, key: &'static str, value: ToolInfo) -> Option<ToolInfo> {
+        self._inner_map.insert(key.into(), value)
+    }
+}
+
+impl From<ToolBTreeMap> for BTreeMap<String, ToolInfo> {
+    fn from(tool_btree: ToolBTreeMap) -> Self {
+        tool_btree._inner_map
+    }
+}
+
+struct StaticToolInfo {
+    /// GitHub repository author
+    pub owner: &'static str,
+
+    /// GitHub repository name
+    pub repo: &'static str,
+
+    /// Executable name inside the .tar.gz or .zip archive
+    pub exe_name: &'static str,
+
+    /// Version tag
+    pub tag: ToolInfoTag,
+
+    pub linux: &'static str,
+    pub macos: &'static str,
+    pub windows: &'static str,
+}
+
+impl From<StaticToolInfo> for ToolInfo {
+    fn from(static_tool_info: StaticToolInfo) -> Self {
+        Self {
```

Comment:
Can we use the `ToolInfo` constructor here for more explicitness?
```suggestion
        ToolInfo {
```

Reply:
I will change the return type as well. I always return `Self` so if the name of the struct changes it's that many less places that also need  change.